flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -934,9 +934,13 @@ function extract_job_id_from_job_submission_return() {
 }
 
 kill_test_watchdog() {
-    local watchdog_pid=$(cat $TEST_DATA_DIR/job_watchdog.pid)
-    echo "Stopping job timeout watchdog (with pid=$watchdog_pid)"
-    kill $watchdog_pid
+  local watchdog_pid=$(cat $TEST_DATA_DIR/job_watchdog.pid)
+      if kill -0 $watchdog_pid > /dev/null 2>&1; then
+           echo "Stopping job timeout watchdog (with pid=$watchdog_pid)"
+           kill $watchdog_pid
+      else
+            echo "Job timeout watchdog (with pid=$watchdog_pid) has already been stopped "
```

Comment:
```suggestion
            echo "No watchdog process with pid=$watchdog_pid present, anymore. No action required to clean the watchdog process up."
```
We should not add knowledge on how this function is called/used. It could be that the function is used in other scenarios in the future (where the process not being present isn't due to the test timed out, for instance). For such a case, the error message would be misleading. I guess, describing the context within the method is good enough. WDYT?

Reply:
I agree with you